### PR TITLE
[docs] Chart.yaml change some fields from optional

### PR DESCRIPTION
**Pull request by sstarcher**
to recommended

### CODE REVIEW

**Comment by sstarcher** [comment]
closes #3802

File: docs/charts.md [hunk 1]
```diff
@@ -49,20 +49,22 @@ kubeVersion: A SemVer range of compatible Kubernetes versions (optional)
...
 engine: gotpl # The name of the template engine (optional, defaults to gotpl)
-icon: A URL to an SVG or PNG image to be used as an icon (optional).
-appVersion: The version of the app that this contains (optional). This needn't be SemVer.
+icon: A URL to an SVG or PNG image to be used as an icon (recommended).
+appVersion: The version of the app that this contains (recommended). This needn't be SemVer.
 deprecated: Whether this chart is deprecated (optional, boolean)
 tillerVersion: The version of Tiller that this chart requires. This should be expressed as a SemVer range: ">2.0.0" (optional)
 ```
 
+Recommended fields if missing will fail linting.
```
Comment (mattfarina) [inline docs/charts.md]: The current linter doesn't fail on the recommended ones. Is this a request to change that behavior? For example, the linter doesn't check for `appVersion`. The code that checks a chartfile is [here](https://github.com/kubernetes/helm/blob/58e4b6ac61938fe056037826225dd19562c0ed9f/pkg/lint/rules/chartfile.go#L35)

Comment (sstarcher) [reply docs/charts.md]: The linter being used by https://github.com/kubernetes/charts does fail when these things are missing.  I have had to add them to charts to get the CI to pass PRs I have submitted.

Comment (sstarcher) [reply docs/charts.md]: Here is a PR that failed until I added `appVersion`
https://circleci.com/gh/kubernetes/charts/6777

**Pull request closed without merging**